src/contracts/Datatoken4.ts

```diff
@@ -131,30 +137,62 @@ export class Datatoken4 extends Datatoken {
    * @param {String} dtAddress Datatoken address
    * @param {String} address User address
    * @param {Boolean} estimateGas if True, return gas estimate
-   * @return {Promise<ReceiptOrEstimate>} transactionId
+   * @return {Promise<ReceiptOrEstimate>} returns the transaction receipt or the estimateGas value
    */
   public async setFileObject<G extends boolean = false>(
     dtAddress: string,
     address: string,
-    fileObject: Bytes,
+    confidentialEVM: boolean = true,
     estimateGas?: G
   ): Promise<ReceiptOrEstimate<G>> {
     if (!(await this.isDatatokenDeployer(dtAddress, address))) {
       throw new Error(`User is not Datatoken Deployer`)
     }
 
     const dtContract = this.getContract(dtAddress)
-    const estGas = await dtContract.estimateGas.setFilesObject(fileObject)
+    const estGas = await dtContract.estimateGas.setFilesObject(this.fileObject)
     if (estimateGas) return <ReceiptOrEstimate<G>>estGas
 
     const trxReceipt = await sendTx(
       estGas,
-      this.signer,
+      confidentialEVM === true ? sapphire.wrap(this.signer) : this.signer,
       this.config?.gasFeeMultiplier,
       dtContract.functions.setFilesObject,
-      fileObject
+      this.fileObject
     )
 
     return <ReceiptOrEstimate<G>>trxReceipt
   }
+
+  /**
+   * getFileObject - It returns the consumer's file object encrypted format.
+   * @param {String} dtAddress datatoken address
+   * @param {Number} serviceId - service identifier
+   * @param {String} providerAddress
+   * @param {Bytes} providerSignature
+   * @param {Bytes} consumerData
+   * @param {Bytes} consumerSignature
+   * @param {String} consumerAddress
+   * @return {Promise<Bytes>} returns file object
+   */
+  public async getFileObject(
+    dtAddress: string,
+    serviceId: number,
```

Comment:
We need to refactor all these occurrences of `serviceID` (including comments).. It definitely is NOT the service id, but the service index.. so `serviceIndex` instead, otherwise we get easily confused

Reply:
right thing, sure, do that now.